### ISSUE TITLE
Question about DND

### ISSUE THREAD

**Issue by zeusshuang** [alexa/avs-device-sdk#1140]
Hi AVS team,

SDK 1.11.  ARM based emabedded linux.

I have a question about DND function introduced in SDK 1.11.
If the DND is enabled from Alexa APP, and I erased all settings in AVS system, and re-register Alexa again.
In Alexa APP, the same device will keep the DND is on in cloud site, but it seems AVS will not sync this DND settings between AVS and Alexa cloud site, so AVS will not know the DND is on. (out of sync with Alexa APP)

Is it an expected behavior in SDK 1.11 ?

Thanks,

**Comment by kjkh** [comment]
Hi @zeusshuang,

Since the AVS cloud is the authoritative source of DND, the device will behave correctly even though it does not know the DND state. So in your scenario, from the cloud perspective DND is on, and the device will behave as such. You are correct that the device is not in a known state since it needs some message (directive, local set command, etc) to determine this. 

Thanks for pointing this out and we will investigate this aspect further. Additionally, it probably makes sense to double check whether an unregistration should reset the DND state as well on the cloud side, so we will also follow up on that.

**Comment by zeusshuang** [comment]
Hi @mavamazon ,

By my steps, I removed all *.db and token information before restart AVS, so from DUT site, it will like a new non-registered DUT. 
Then I registered it with same account and ID, the Alexa cloud site will treat it as the same device, so the DND will keep Enabled while I check this device from Alexa APP. 
But from DUT site, it will not know it's DND enabled case.

**Comment by dhpp** [comment]
Hi @zeusshuang, what do you mean about the DUT site?  Please note that DND is an Alexa feature, so if a device connects to AVS with the same credentials as before, you should expect the same DND setting to be present (unless the device was explicitly de-registered).

**Comment by zeusshuang** [comment]
Hi @dhpp ,

Yes, I understand that DND is an Alexa feature.
However, it will out of sync after AVS is deregistered from Alexa APP or log-off from AVS site.
I did this test:
Enabled DND from Alexa APP, then do deregister from APP. The authentication data will be removed from cloud site, but AVS could not know the de-register status (it's another issue).  Then, I removed all settings and register AVS again (same device information), after authenticated, I saw the DND is still enabled from Alexa APP (it seems Alexa cloud site will keep the previous setting), but AVS will not receive the DND enabled message until I disable / enable it again from APP.
